staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller.go

```diff
@@ -535,7 +535,7 @@ func CheckRequest(quotas []corev1.ResourceQuota, a admission.Attributes, evaluat
 		return quotas, err
 	}
 	if len(scopesHasNoCoveringQuota) > 0 {
-		return quotas, fmt.Errorf("insufficient quota to match these scopes: %v", scopesHasNoCoveringQuota)
+		return quotas, admission.NewForbidden(a, fmt.Errorf("insufficient quota to match these scopes: %v", scopesHasNoCoveringQuota))
```

Comment:
I agree this looks like it should be handled as a quota shortfall, meaning a forbidden error (same as line 527 above).

Is there a test covering this? can we add an error type check to the test?

Reply:
There currently isn't but I can add one to catch this use case.

Reply:
It appears as though this whole controller is lacking tests so this might take a bit longer to get in place.